### ISSUE TITLE
Define standard package.json structure for extensions

### ISSUE THREAD

**Issue by stevejr** [lensapp/lens#1286]
Define a standard package.json structure for Lens extensions. The below example contains mandatory fields with example content:

```
{
  "name": "@k8slens/extensions-sample",
  "version": "0.0.1",
  "author": "Mirantis, Inc <info@k8slens.dev> (https://k8slens.dev)",
  "publisher": "k8slens",
  "maintainers": [
    {
      "name": "k8slens-bot",
      "email": "info@k8slens.dev"
    }
  ],
  "description": "Lens extension sample",
  "repository": {
    "type" : "git",
    "url" : "https://github.com/lensapp/lens-extension-samples.git",
    "directory": "sample"
  },
  "homepage": "https://github.com/lensapp/lens-extension-samples",
  "bugs": {
    "url": "https://github.com/lensapp/lens-extension-samples/issues"
  },
  "engines": {
    "lens": "^4.0.0"
  },
  "license": "MIT",
  "keywords": [
    "example",
    "extension",
    "k8slens",
    "lens",
    "mirantis",
    "tutorial"
  ],
  "main": "dist/main.js",
  "renderer": "dist/renderer.js",
  "scripts": {
    "build": "webpack --config webpack.config.js",
    "dev": "npm run build --watch"
  },
  "dependencies": {
    "react-open-doodles": "^1.0.5"
  },
  "devDependencies": {
    "@k8slens/extensions": "^4.0.0-alpha.2",
    "ts-loader": "^8.0.4",
    "typescript": "^4.0.3",
    "@types/react": "^16.9.35",
    "@types/node": "^12.0.0",
    "webpack": "^4.44.2",
    "webpack-cli": "^3.3.11"
  }
}

It is expected that all of the above fields would be present.

**Comment by stefcameron** [comment]
# Required vs Optional

Do we really want to __require__ all those fields? Do you _really_ have to have:
- `keywords`
- `dependencies` and `devDependencies`: You probably would want to have at least one of those to reference the `@k8slens/extensions` dependency, but since that package only contains typings, and the actual values are injected as globals into the Extension scope, you actually don't need this package, especially if you aren't using TypeScript, provided you use the documentation to know what's in the global scope.
- `scripts`: Would be strange not to have this, but technically, what does this have to do with extensions?
- `license`, `bugs`, `homepage`, `repository`: Unless you intend to pull this into an Extension Page in a Marketplace View in Lens some day, but still, these feel like they should be optional, not required. If no license, you can assume "unlicensed"
- `description`: You could argue this is also optional. The `name` is the most critical part. If you have this Marketplace with pages for each Extension, similar to VSCode, then as an extension dev, you should want to include a description, but will this affect the functioning of the extension? Probably not.
- `author`, `maintainers`: This is just metadata...

I would say we're left with these required fields:
- `name`
- `version`
- `publisher`
- `engines.lens`
- `main`
- `renderer`

# lensExtension in package.json

I would also strongly recommend that `publisher`, `main`, and `renderer`, which are all Lens extension-specific, should be moved into a common `lensExtension` section to be grouped together.

Otherwise, if you add more Lens-specific properties in the future, you'll end-up further polluting the standard [package.json](https://docs.npmjs.com/cli/v6/configuring-npm/package-json) structure with non-package manager-related properties, all related to one another, scattered throughout the whole file. It could be become a mess...

You might also end-up in a name collision, which you could avoid now, for the future, by moving those properties into a single object.

This is common practice in the greater JS community, see [ESLint](https://eslint.org/docs/user-guide/configuring#configuration-file-formats) (key: `eslintConfig`), [Prettier](https://prettier.io/docs/en/configuration.html) (key: `prettier`), [Babel](https://babeljs.io/docs/en/config-files) (key: `babel`), etc.

# Minimum package.json

I would propose that this is the minimum required package.json an extension would need:

```json
{
  "name": "@k8slens/extensions-sample",
  "version": "0.0.1",
  "engines": {
    "lens": "^4.0.0"
  },
  "lensExtension": {
    "publisher": "k8slens",
    "main": "dist/main.js",
    "renderer": "dist/renderer.js"
  }
}
```

**Comment by stefcameron** [comment]
Granted `lensExtension` could still result in a collision with some new standard package.json property later on, but that seems less likely than the alternative.

**Comment by chenhunghan** [comment]
I think package.json structure/scheme can be included in the generator template, I linked this issue to https://github.com/lensapp/lens/issues/1296

**Comment by nevalla** [comment]
I agree that we should only require that minimum set of fields that are required to run extensions properly. So extension developers can easily develop/test extensions with the Lens app.
